plugins/rebalance/rest.go

```diff
@@ -0,0 +1,364 @@
+		return
+	}
+
+	rebalanceinfo.Volname = volname
+	rebalanceinfo.Status = GfDefragStatusStarted
+	setcommithash(rebalanceinfo)
+	rebal, err := createRebalanceInfo(rebalanceinfo)
+	if err != nil {
+		logger.WithError(err).Error("failed to create Rebalance info")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Ctx.Set("rinfo", rebal)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	if err = txn.Ctx.Get("rinfo", &rebal); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get rebalanceinfo", api.ErrCodeDefault)
+		return
+	}
+
+	txn.Ctx.Logger().WithField("volname", rebal.Volname).Info("rebalance started")
+	resp := rebalanceStartResp(rebal)
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, resp)
+}
+
+func rebalanceStartResp(v *RebalanceInfo) *api.RebalanceStartResp {
+	return (*api.RebalanceStartResp)(rebalanceStartInfoResp(v))
+}
+
+func rebalanceStop(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	rebalanceinfo := new(RebalanceInfo)
+
+	//collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	//Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount == 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distributed volume or contain only 1 brick",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// Check remove brick operation is running
+	//TODO
+
+	//A simple transaction to stop rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebal-stop.Commit",
+			Nodes:  txn.Nodes,
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	rebalanceinfo.Volname = volname
+	rebalanceinfo.Status = GfDefragStatusStopped
```

Comment:
Status and cmd are not the same and it looks like they may be used interchangeably here.